### PR TITLE
feat(ui): finding → analysis trace entry points and the agent-trace viewer

### PR DESCRIPTION
**Pull request by GabrielDrapor**
Makes the traces reachable. Four entry points:

- **Runs table** — the Finding ID becomes clickable, styled like the Run ID and Trace ID beside it, opening the analysis trace in the same in-page viewer panel the page already uses. It stays plain text while the analysis is running, when tracing failed, and for findings analysed before tracing existed; the hover title says which.
- **Alert panel** — a **View analysis trace** button swaps the viewer to the analysis trace; **← Back to trace** returns.
- **Sidebar** — every finished assistant reply gets a **View trace** link next to its token count, opening the turn's trace in a sheet over the current page. Links appear only when `trace_status = available`.
- **Viewer** — an **Agent** badge in the header, and span names that read as what the agent did.

**The invariant to review hardest:** analysis traces never appear in the Traces list, and the viewer for customer traces is unchanged. The snapshot test of the user-trace viewer is taken in its own commit *before* the changes, so the diff proves it stayed byte-identical.

Superseded by #2082, which now carries the link to issue 2064. Part of #2058.

### CODE REVIEW

File: frontend/ui/src/features/ai-assistant/components/message-list.tsx [hunk 1]
```diff
@@ -539,13 +578,27 @@ export function MessageList({ messages, sessionStreaming = false }: MessageListP
...
       <div ref={innerRef}>
-        {messages.map((msg) => {
+        {messages.map((msg, index) => {
           if (msg.role === "tool_step" && msg.toolStep) {
+            // The step's own turn hasn't produced its trace-carrying assistant
+            // bubble yet until that bubble appears later in the list — its
+            // traceId is the one this tool step's span belongs to.
+            const turnTraceId = messages
+              .slice(index + 1)
+              .find((m) => m.role === "assistant")?.traceId;
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/ai-assistant/components/message-list.tsx]: Valid on both counts. A tool-only run produces no assistant bubble, so the forward search ran past the turn boundary and attached the step to the *next* turn's trace — a different trace that does not contain that span. The search now stops at the next user message, and the link is gated on `traceStatus === "available"`, matching the turn's own "View trace". A new message-list test file covers the correct case, the boundary case, and each non-available status.

File: frontend/ui/src/features/ai-assistant/components/agent-trace-sheet.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        if (!o) onClose();
+      }}
+    >
+      <DrawerContent width="w-[min(90vw,1100px)]" className="p-0">
+        <DrawerTitle className="sr-only">Trace</DrawerTitle>
+        {traceId && (
+          <LayoutContext.Provider
+            value={{ ...layout, registerAiHost: () => () => {}, aiPanelOpen: false }}
+          >
+            <TraceViewerPanel
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/ai-assistant/components/agent-trace-sheet.tsx]: Valid — nesting a fixed 70%-viewport overlay inside DrawerContent paints over the page instead of filling it. An explicit `embedded` prop drops the fixed positioning. Worth noting the non-embedded class string is now written out in full rather than composed: the first attempt reordered the classes identically-but-not-textually and the user-trace snapshot test caught it, which is exactly what that test is for.

File: frontend/ui/src/features/ai-assistant/components/agent-trace-sheet.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      open={!!traceId}
+      onOpenChange={(o) => {
+        if (!o) onClose();
+      }}
+    >
+      <DrawerContent width="w-[min(90vw,1100px)]" className="p-0">
+        <DrawerTitle className="sr-only">Trace</DrawerTitle>
+        {traceId && (
+          <LayoutContext.Provider
+            value={{ ...layout, registerAiHost: () => () => {}, aiPanelOpen: false }}
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/ai-assistant/components/agent-trace-sheet.tsx]: Valid, and slightly worse than described: `aiPanelOpen` was hard-coded false but `setAiPanelOpen` still reached the outer panel, so the button opened an assistant the user cannot see from inside the sheet. `embedded` hides the control rather than neutering the setter — the same context still serves the Alert badge's assistant handoff, which must keep working.

File: frontend/ui/src/features/detectors/components/detector-runs-table.tsx [hunk 4]
```diff
@@ -94,9 +101,27 @@ export function DetectorRunsTable({ rows, onTraceClick, onRunClick }: DetectorRu
...
+                  </button>
                 ) : (
                   <span
-                    title={findingId}
+                    title={
+                      run.execution_trace_status == null
+                        ? `${findingId} — no analysis trace (analysis ran before tracing was enabled)`
+                        : run.execution_trace_status === "pending"
+                          ? `${findingId} — analysis trace is being recorded`
+                          : `${findingId} — analysis trace unavailable`
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/detectors/components/detector-runs-table.tsx]: Valid — absent execution status means different things depending on whether an analysis ran at all, and `rca_status` distinguishes them. A never-analyzed finding now reads "not analyzed", a failed one says so, and only a completed RCA with no trace claims it predates tracing.

File: frontend/ui/src/features/ai-assistant/components/ai-assistant-panel.tsx [hunk 5]
```diff
@@ -262,6 +272,13 @@ export function AiAssistantPanel({
           }
         />
       </div>
+      {projectId && (
+        <AgentTraceSheet
+          projectId={projectId}
+          traceId={openTrace?.traceId ?? null}
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/ai-assistant/components/ai-assistant-panel.tsx]: Correct for this PR — the panel holds `spanId` in state but stops there. Threading it through the sheet into the viewer is #2074, the next PR in the stack, which also fixes the effect race that kept the deep link from applying. Split that way so the sheet's plumbing and the viewer's selection logic review separately.

File: frontend/ui/src/features/ai-assistant/components/message-list.tsx [hunk 6]
```diff
@@ -386,6 +395,15 @@ function ToolStepItem({ step, isActive }: { step: ToolCallStep; isActive: boolea
                 </pre>
               </div>
             )}
+            {step.spanId && onOpenSpan && (
+              <button
+                type="button"
+                className="text-muted-foreground/60 hover:underline"
+                onClick={() => onOpenSpan(step.spanId!)}
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/ai-assistant/components/message-list.tsx]: Same as the comment above: the id reaches the panel but is not passed on until #2074, which threads it through and makes the viewer honour it.

File: frontend/ui/src/app/api/projects/[projectId]/detectors/[detectorId]/runs/route.ts [hunk 7]
```diff
@@ -83,12 +85,19 @@ export async function GET(req: NextRequest, { params }: RouteParams) {
...
           });
-          const statusByFinding = new Map(rcas.map((r) => [r.findingId, r.status]));
+          const byFinding = new Map(rcas.map((r) => [r.findingId, r]));
           for (const r of runs as Array<Record<string, unknown>>) {
             if (typeof r.finding_id === "string") {
-              r.rca_status = statusByFinding.get(r.finding_id) ?? null;
+              const rca = byFinding.get(r.finding_id);
+              r.rca_status = rca?.status ?? null;
+              r.execution_trace_id = rca?.latestExecution?.traceId ?? null;
+              r.execution_trace_status = rca?.latestExecution?.traceStatus ?? null;
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/app/api/projects/[projectId]/detectors/[detectorId]/runs/route.ts]: Real, and it is the reason `trace_status` is a four-state enum rather than a boolean. While a retry is pending the pointer does still name the prior attempt, so its trace is offered — that trace exists and is the newest complete analysis, which is the honest answer until a newer one finishes. What the design does not want is offering a link to a trace that was never exported, which the `available` gate prevents. Making the pointer move before success would mean advertising an execution that has produced nothing.

File: frontend/ui/src/features/traces/components/TraceViewerPanel.tsx [hunk 8]
```diff
@@ -591,15 +669,16 @@ export function TraceViewerPanel({
                     />
                   ) : error || !trace ? (
                     <div className="flex h-full items-center justify-center">
-                      {source === "detector" &&
+                      {(effectiveSource === "detector" || effectiveSource === "agent") &&
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/traces/components/TraceViewerPanel.tsx]: The pending window is bounded, not indefinite: `isSelfTracePending` compares `runTimestamp` against a 60s window and past it a miss reads as failed rather than pending. You are right that an *absent* `runTimestamp` stays pending forever — deliberate, since not knowing when a run started is no evidence its export failed, but the runs table always passes one. The agent-trace sheet does not, and reaches traces by an id the API already resolved, so a 404 there is a different condition; worth revisiting if that path starts 404ing in practice.

File: frontend/ui/src/features/traces/components/TraceViewerPanel.tsx [hunk 9]
```diff
@@ -433,13 +507,17 @@ export function TraceViewerPanel({
...
+                      // Uses the effective id/source, not the raw props, so
+                      // popping out while viewing the RCA's agent trace (via
+                      // "View analysis trace") reopens THAT trace, not the
+                      // original one underneath it.
                       extraParams:
-                        source === "detector"
-                          ? { traceId, fullscreen: "1", source }
-                          : { traceId, fullscreen: "1" },
+                        effectiveSource === "detector" || effectiveSource === "agent"
+                          ? { traceId: effectiveTraceId, fullscreen: "1", source: effectiveSource }
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/traces/components/TraceViewerPanel.tsx]: Correct as stated, and it is why the popout is not offered from the sheet — the agent-trace sheet renders the viewer without the new-tab control. The finding-ID and Alert-panel entries open the trace in the detector page's own in-page viewer, which passes `source=agent`, rather than routing through the Traces page. Making that page source-aware is the open question about listing internal traces (design B4a), not something this PR settles.

File: frontend/ui/src/features/traces/components/TraceViewerPanel.tsx [hunk 10]
```diff
@@ -392,6 +446,26 @@ export function TraceViewerPanel({
                 Alert
               </button>
             )}
+            {hasRca && agentTrace && !viewingAgentTrace && (
+              <Button
+                variant="ghost"
+                size="sm"
+                className="h-7 px-2 text-xs"
+                onClick={() => setViewingAgentTrace(true)}
```
[the remark this answers is not in the document]
Comment (GabrielDrapor) [reply frontend/ui/src/features/traces/components/TraceViewerPanel.tsx]: The reset does key on the trace: its dependency is `traceId`, and `effectiveTraceId` only differs from it while the Alert-panel swap is showing the agent trace — which sets `viewingFinding` and re-renders through the same path. Collapse state resets with it. View mode is deliberately kept: it is a user preference about how to read a trace, not per-trace state, and resetting it would drop someone out of Timeline every time they swapped. The Detectors tab case is real but not reachable from here — the swap control lives in the Alert panel, which the Detectors tab does not render.
